app/bookmark.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2018 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class Bookmark(object):
+  """
+  This bookmark object is used as a marker for different places in a
+  text document. Note that because text buffer lines index at 0, all
+  references to rows also assume that 0 is the first line.
+  """
+  def __init__(self, beginRow, endRow, data={}):
+    """
+    Creates a bookmark located at the specified rows. This will automatically
+    cast the passed in rows into integers.
+    Args:
+      beginRow (int): The line that the bookmark starts on (inclusive).
+      endRow (int): The line that the bookmark ends on (inclusive).
```

Comment:
Is there a (type) specifier for 'thing that converts to int'? i.e. it's only kinda accurate that an int is expected -- sure an int is preferred, but it looks like anything that can convert to an int can be passed in. 
Or, maybe truly expect/require ints and not do the int() conversion below?
(Either way is okay with me, just making a note that the docs and code don't seem to line up).

Reply:
I think the python nomenclature would be 'int-like'.